### PR TITLE
Личный проект: большой взрыв

### PR DESCRIPTION
**Pull request by dpsnow**
---
:mortar_board: [Личный проект: большой взрыв](https://up.htmlacademy.ru/ecmascript/8/user/574945/tasks/5)

:boom: https://htmlacademy-ecmascript.github.io/574945-big-trip-8/2/

### CODE REVIEW

File: src/data.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+
+const listFilter = [
+  {name: `Everything`, checked: true},
+  {name: `Future`},
+  {name: `Past`}
+];
+
+const listTripPoint = [
```
Comment (rkhaslarov) [inline src/data.js]: No need to use prefix LIST, it's better to name array as plural form, tripPoints or filters

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import {listFilter, listTripPoint} from './data.js';
+import {getRandomInt, insertHtmlToElement, getNewListTripPoints} from './utils.js';
+
+import {createFilter} from './templateFilter.js';
+import {createTripPoint} from './templateTripPoint.js';
+
+const COUNT_TRIP_POINT_FIRST = 4;
```
Comment (rkhaslarov) [inline src/main.js]: COUNT_TRIP_POINT_FIRST didn't get what does it mean?=)

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+const renderTripPoint = (container, parametrsTripPoint) => {
+  const fragment = document.createDocumentFragment();
+  const tripPoints = parametrsTripPoint.map(createTripPoint);
+  tripPoints.forEach((element) => {
+    insertHtmlToElement(element, fragment);
+  });
+  container.appendChild(fragment);
+};
```
Comment (rkhaslarov) [inline src/main.js]: Don't you think that renderTripPoint and renderFilters a bit the same? ) Don't forget the rule of duplication.

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+const init = () => {
+  const firstTripPoint = getNewListTripPoints(COUNT_TRIP_POINT_FIRST, listTripPoint);
+  renderFilters(filtersContainer, listFilter);
+  renderTripPoint(tripPointContainer, firstTripPoint);
+
+  filtersContainer.addEventListener(`click`, (evt) => {
+    if (evt.target.nodeName === `INPUT`) {
+      tripPointContainer.innerHTML = ``;
+      const newCounttripPoint = getRandomInt(1, MAX_TRIP_POINT);
```
Comment (rkhaslarov) [inline src/main.js]: newCounttripPoint? name should be consistent and according to camelCase

File: src/utils.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+const parser = new DOMParser();
+
+const getRandomInt = (min, max) => Math.floor(Math.random() * Math.floor((max - min) + 1) + min);
+
+const insertHtmlToElement = (html, container) => {
```
Comment (rkhaslarov) [inline src/utils.js]: JSDoc as below

File: src/utils.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+const parser = new DOMParser();
+
+const getRandomInt = (min, max) => Math.floor(Math.random() * Math.floor((max - min) + 1) + min);
```
Comment (rkhaslarov) [inline src/utils.js]: JSDoc as below

File: src/main.js (hunk 3 above)
Comment (rkhaslarov) [reply src/main.js]: Probably it's better to create function in utils kind of render function

File: src/data.js (hunk 1 above)
Comment (dpsnow) [reply src/data.js]: done.

File: src/main.js (hunk 2 above)
Comment (dpsnow) [reply src/main.js]: I changed the name and value to `NUMBER_TRIP_POINTS_ON_PAGE`
This means the number of trip points when the page loads. =)
`Используя функцию из пункта 4 отрисуйте семь одинаковых событий.`

File: src/main.js (hunk 3 above)
Comment (dpsnow) [reply src/main.js]: I created new function `renderElements`

File: src/main.js (hunk 4 above)
Comment (dpsnow) [reply src/main.js]: done

File: src/utils.js (hunk 6 above)
Comment (dpsnow) [reply src/utils.js]: done

File: src/utils.js (hunk 5 above)
Comment (dpsnow) [reply src/utils.js]: done

**Pull request merged**